packages/assets-controllers/src/MultichainAssetsController/MultichainAssetsController.ts

```diff
@@ -0,0 +1,561 @@
+import type {
+  AccountsControllerAccountAddedEvent,
+  AccountsControllerAccountAssetListUpdatedEvent,
+  AccountsControllerAccountRemovedEvent,
+  AccountsControllerListMultichainAccountsAction,
+} from '@metamask/accounts-controller';
+import {
+  BaseController,
+  type ControllerGetStateAction,
+  type ControllerStateChangeEvent,
+  type RestrictedControllerMessenger,
+} from '@metamask/base-controller';
+import { isEvmAccountType } from '@metamask/keyring-api';
+import type {
+  AccountAssetListUpdatedEventPayload,
+  CaipAssetType,
+  CaipAssetTypeOrId,
+} from '@metamask/keyring-api';
+import type { InternalAccount } from '@metamask/keyring-internal-api';
+import { KeyringClient } from '@metamask/keyring-snap-client';
+import type {
+  GetPermissions,
+  PermissionConstraint,
+  SubjectPermissions,
+} from '@metamask/permission-controller';
+import type {
+  GetAllSnaps,
+  HandleSnapRequest,
+} from '@metamask/snaps-controllers';
```

Comment:
It looks like `snaps-controllers` is a dev dependency, but do we need to also add it as a peer dependency?